src/main/src/main/java/org/geoserver/catalog/ResourcePool.java

```diff
@@ -1792,11 +1792,11 @@ public void clear(StyleInfo info) {
      * @return A reader for the style.
      */
     public BufferedReader readStyle( StyleInfo style ) throws IOException {
-        File styleFile = dataDir().findStyleSldFile(style);
-        if( styleFile == null ) {
-            throw new IOException( "No such file: " + style.getFilename() );
+        Resource styleResource = dataDir().style(style);
```

Comment:
I don't think it should scare you too much, just updating to use a non-deprecated method for style body retrieval that uses Resources under the hood. The same method is already being used in a number of other places such as writeStyle in the same class and it's used numerous times in StyleResource.java in the catalog. The method should already be pretty well tested.